File: 260.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        xorsum = 0
        for num in nums:
            xorsum ^= num
        
        lsb = xorsum & (-xorsum)

        ans1, ans2 = 0, 0
        for num in nums:
            if num & lsb:
                ans1 ^= num
            else:
                ans2 ^= num

        return [ans1, ans2]

File: go/260.go
package main

func singleNumber(nums []int) []int {
	x := 0
	for _, num := range nums {
		x ^= num
	}
	i := x & -x
	x1, x2 := 0, 0
	for _, num := range nums {
		if num&i == 0 {
			x1 ^= num
		} else {
			x2 ^= num
		}
	}
	return []int{x1, x2}
}

func main() {

}
